debezium-server-redis/src/test/java/io/debezium/server/redis/RedisStreamHeartbeatDisabledIT.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.server.redis;
+
+import static org.junit.Assert.assertTrue;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.Test;
+
+import io.debezium.connector.postgresql.connection.PostgresConnection;
+import io.debezium.util.Testing;
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusIntegrationTest;
+import io.quarkus.test.junit.TestProfile;
+
+import redis.clients.jedis.HostAndPort;
+import redis.clients.jedis.Jedis;
+
+/**
+ * Integration test that verifies heartbeat messages are stored in Redis Stream when skip.heartbeat.messages=false
+ *
+ * @author Yossi Shirizli
+ */
+@QuarkusIntegrationTest
+@TestProfile(RedisStreamHeartbeatDisabledTestProfile.class)
+@QuarkusTestResource(RedisTestResourceLifecycleManager.class)
+public class RedisStreamHeartbeatDisabledIT {
+
+    /**
+     * Test that heartbeat messages are stored when skip.heartbeat.messages=false
+     */
+    @Test
+    public void testHeartbeatMessagesStored() throws Exception {
+        Testing.Print.enable();
+
+        // Get a connection to Redis
+        Jedis jedis = new Jedis(HostAndPort.from(RedisTestResourceLifecycleManager.getRedisContainerAddress()));
+
+        // Create a table and insert data to trigger the CDC process
+        final PostgresConnection connection = TestUtils.getPostgresConnection();
+        Testing.print("Creating new redis_test table and inserting 2 records to it");
+        connection.execute(
+                "CREATE TABLE inventory.redis_test (id INT PRIMARY KEY)",
+                "INSERT INTO inventory.redis_test VALUES (1)",
+                "INSERT INTO inventory.redis_test VALUES (2)");
+        connection.close();
+
+        // Wait for some time to allow heartbeats to be generated
+        Testing.print("Waiting for heartbeats to be generated...");
+        Thread.sleep(3000);
```

Comment:
Could you please rewrite the test to not use `Thread.sleep()` but `Awaitility` that would wait (with timeout) till something is in heartbeat stream?

Reply:
Thanks, done.